### DISCUSSION TITLE
Support creating a "linked note" for one-way synced remote calendars

### DISCUSSION THREAD

**Discussion by zachdaniel** [davish/obsidian-full-calendar#352] [Feature Requests]
Although at some point the idea is to do two-way sync, a short term but very useful thing could be to support creating a note for an event. This could possibly be "on-demand", e.g if no matching note is found for a given synced event than nothing happens, and then if you click on an event you are given the option to create a note for it.

Ultimately, two-way sync pretty much covers this case, but this could be a good stepping stone along the way.

Thanks for your hard work on this! Love this extension.

**Comment by davish** [comment]
This could pretty easily be an action in #7. I do worry about creating a duplicate event. @zachdaniel would there be a flow that would make sense to you? Perhaps a frontmatter field `hideIcsEvent: <ics_id>` which, when present, would suppress the event on the remote calendar from showing up in the calendar view.

Or would a duplicate event in the view be acceptable here?

**Comment by zachdaniel** [comment]
🤔 I think maybe `icsEvent: <ics_id>` would be better, because the note shouldn't really be authoritative on the date/time, but the calendar event would be. So all you'd really be doing is connecting it so when you click on the event it takes you to the proper note?

**Comment by zachdaniel** [comment]
For example, I'm taking notes in a meeting, and then after the fact I want to say something like "these are the notes for X meeting on my calendar"

**Comment by pcause** [comment]
A few comments:

- there will be calendars that can't be two way synced.  imagine I download a baseball team calendar and want to "delete" a few home games because I won't be able to go to the game.  Or a Holiday calendar and I don't have the holiday off and need to work.  there are many calendars we need to see that we can't change.
- i think this is best added along with the idea of having multiple events in a document.  otherwise we clutter up things with lots of tiny documents.  the document created for a note cane just be an item that indicates it is a note for the event and not an event, or perhaps an override for the calendar event.

I think tying this into the multiple events in a document is the simplest way to implement.  For example, if there is a pseudo event created to delete an event and the user decides they want to restore they can just edit the event directly in the document.

**Comment by midnightmastermind** [comment]
Any update on this?

**Comment by alancunha26** [comment]
Updates?

**Comment by davish** [comment]
@alancunha26, creating a mapping from any file in a Vault back to an event would not be particularly difficult in the current code but it would be woefully inefficient and slow as users add more files to their vaults. I'm currently working on a global cache for events that can also store the location of associated notes in an efficient way. You can follow PR #204 for updates on that work. This issue will not be completed until after #204 is merged.

**Comment by sunt05** [comment]
Thanks for the work! With #204 merged, will this feature be available soon?

**Comment by FetchFast** [comment]
+1. Desperate for some functionality like this. Very important feature. 

Everybody else: what's your workflow workaround while we wait?
Do you work in a team? Did you ask the person providing the calendar to provide it in a different format?

I have a Google Calendar widget open on my phone for speed and then just have to name events manually

**Reply by FetchFast** [reply]
I'm not getting an activation at bountysource, so could someone start a bounty for this, and I'll add $10

**Comment by dannydi12** [comment]
This is the reason I downloaded the plugin too! The README says:
> Each event is stored as a separate note with special frontmatter so you can take notes, form connections and add context to any event on your calendar.

Why would I want to maintain a whole separate calendar locally within Obsidian? I want to store notes related to my google calendar events

**Comment by egoipse** [comment]
Still no news about this issue?

**Comment by planefun2** [comment]
Really wish we had this @davish 🥇

**Comment by pgmillon** [comment]
Really need this feature, having a read only calendar makes the plugin barely useful since I have to explicitly create a note for each of my meeting and copy/paste the title, attendees, etc
